cluster/gce/gci/append_or_replace_prefixed_line_test.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gci
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestAppendOrReplacePrefix(t *testing.T) {
+	testCases := []struct {
+		desc                string
+		prefix              string
+		suffix              string
+		initialFileContents string
+		want                string
+	}{
+		{
+			desc:   "simple string and empty file",
+			prefix: "hello",
+			suffix: "world",
+			want: `helloworld
+`,
+		},
+		{
+			desc:                "simple string and non empty file",
+			prefix:              "hello",
+			suffix:              "world",
+			initialFileContents: "jelloworld",
+			want: `jelloworld
+helloworld
+`,
+		},
+		{
+			desc:                "simple string and file already contains prefix",
+			prefix:              "hello",
+			suffix:              "world",
+			initialFileContents: "helloworld",
+			want: `helloworld
+`,
+		},
+		{
+			desc:   "string with quotes and $ and empty file",
+			prefix: "'\"$argon2id$v=19\"'",
```

Comment:
nit: use a raw string to avoid escaping (here and below)

Reply:
Done.